pkg/codegen/go/gen_program_test.go

```diff
@@ -31,56 +41,72 @@ func TestCollectImports(t *testing.T) {
-	files, err := ioutil.ReadDir(testdataPath)
+	path := filepath.Join(testdataPath, testFile)
+	contents, err := ioutil.ReadFile(path)
+	assert.NoErrorf(t, err, "could not read %v: %v", path, err)
+
+	parser := syntax.NewParser()
+	err = parser.ParseFile(bytes.NewReader(contents), filepath.Base(path))
 	if err != nil {
-		t.Fatalf("could not read test data: %v", err)
+		t.Fatalf("could not read %v: %v", path, err)
+	}
+	if parser.Diagnostics.HasErrors() {
+		t.Fatalf("failed to parse files: %v", parser.Diagnostics)
 	}
 
-	for _, f := range files {
-		if filepath.Base(f.Name()) != testFile {
-			continue
-		}
-
-		path := filepath.Join(testdataPath, f.Name())
-		contents, err := ioutil.ReadFile(path)
-		if err != nil {
-			t.Fatalf("could not read %v: %v", path, err)
-		}
-
-		parser := syntax.NewParser()
-		err = parser.ParseFile(bytes.NewReader(contents), f.Name())
-		if err != nil {
-			t.Fatalf("could not read %v: %v", path, err)
-		}
-		if parser.Diagnostics.HasErrors() {
-			t.Fatalf("failed to parse files: %v", parser.Diagnostics)
-		}
+	program, diags, err := hcl2.BindProgram(parser.Files, hcl2.PluginHost(utils.NewHost(testdataPath)))
+	if err != nil {
+		t.Fatalf("could not bind program: %v", err)
+	}
+	if diags.HasErrors() {
+		t.Fatalf("failed to bind program: %v", diags)
+	}
 
-		program, diags, err := hcl2.BindProgram(parser.Files, hcl2.PluginHost(utils.NewHost(testdataPath)))
-		if err != nil {
-			t.Fatalf("could not bind program: %v", err)
-		}
-		if diags.HasErrors() {
-			t.Fatalf("failed to bind program: %v", diags)
-		}
+	g := &generator{
+		program:             program,
+		jsonTempSpiller:     &jsonSpiller{},
+		ternaryTempSpiller:  &tempSpiller{},
+		readDirTempSpiller:  &readDirSpiller{},
+		splatSpiller:        &splatSpiller{},
+		optionalSpiller:     &optionalSpiller{},
+		scopeTraversalRoots: codegen.NewStringSet(),
+		arrayHelpers:        make(map[string]*promptToInputArrayHelper),
+	}
+	g.Formatter = format.NewFormatter(g)
+	return g
+}
 
-		g := &generator{
-			program:             program,
-			jsonTempSpiller:     &jsonSpiller{},
-			ternaryTempSpiller:  &tempSpiller{},
-			readDirTempSpiller:  &readDirSpiller{},
-			splatSpiller:        &splatSpiller{},
-			optionalSpiller:     &optionalSpiller{},
-			scopeTraversalRoots: codegen.NewStringSet(),
-			arrayHelpers:        make(map[string]*promptToInputArrayHelper),
-		}
-		g.Formatter = format.NewFormatter(g)
-		return g
+func goCheck(t *testing.T, path string) {
+	dir := filepath.Dir(path)
+	ex, err := executable.FindExecutable("go")
+	assert.NoError(t, err)
+	_, err = ioutil.ReadFile("go.mod")
+	if os.IsNotExist(err) {
+		defer func() {
+			// If we created the module, we also remove the module
+			err = os.Remove(filepath.Join(dir, "go.mod"))
+			assert.NoError(t, err)
+			err = os.Remove(filepath.Join(dir, "go.sum"))
+			assert.NoError(t, err)
+		}()
+		err = integration.RunCommand(t, "generate go.mod",
```

Comment:
Similarly to the compile check for SDK codegen, this needs `go mod edit` pointing to in-repo Go SDK with a replace directive.